handshake.md

```diff
@@ -0,0 +1,583 @@
+authenticate all further outgoing and incoming Cable Wire Protocol messages.
+
+A successful Cable Handshake will resemble the following exchange of
+*handshake messages*:
+
+```
+INITIATOR                                 RESPONDER         STEP
+========================================================================
+  [MAJOR].[MINOR] version ---------------------->           (1)
+  
+  <---------------------- [MAJOR].[MINOR] version           (2)
+
+  Noise ephemeral key ------------------------->            (3)
+
+  <------------ Noise ephemeral key + static key            (4)
+
+  Noise static key ---------------------------->            (5)
+
+  <----- Bidirectional encrypted messages ----->            (6)
+```
+*Figure 1.0      Handshake message exchange overview*
+
+Steps 1 and 2 are part of the Version Exchange phase of the handshake, and
+steps 3-5 belong to the Noise Handshake phase. Upon completion of steps
+1-5, the protocol enters into Post-Handshake Operation phase in step 6.
+
+## 2. Key concepts
+
+### 2.1 Initiator and responder
+In a Cable Handshake, one host takes on the role of *initiator* and the other
+takes on the role of *responder*. This is a requirement of the *Noise Protocol
+Framework*, which is discussed further in the next subsection.
+
+Since different transports have different properties, a single rule cannot be
+provided as to which host takes on which role. However, for transport protocols
+where there are well-defined *client* and *server* roles, such as TCP/IP,
+implementations SHOULD regard the client as the initiator and the server as the
+responder.
+
+The role chosen only decides the order of messages sent during the handshake
+process and does not affect how the Cable Wire Protocol operates.
+
+### 2.3 Noise
+The Noise Protocol Framework is a framework for building cryptographically
+secure protocols. It is not a protocol in itself, but rather a framework for
+producing protocols specific to an application's cryptographic needs.
+
+Cable uses Revision 34 (2018-07-11) of Noise. It can be found mirrored in this
+repository and is referenced in this document as the "Noise specification"
+[[2](#ref2)]. It is mirrored here because its current URL is not locked to a
+specific version, and may change: https://noiseprotocol.org/noise.html
+
+This document uses terminology defined in the Noise specification. Names and
+specification sections will be *italicized*, and references to functions or
+other pseudocode defined in the Noise specification will appear in `code
+blocks`.
+
+#### 2.3.1 Protocol name
+A specific Noise protocol is defined by a unique string -- the *protocol name*
+-- that describes i) the messages exchanged by each host, ii) whether
+pre-shared keys are used (and when they are incorporated), and iii) the suite
+of cryptographic primitives to be used.
+
+The *protocol name* used by Cable is `Noise_XXpsk0_25519_ChaChaPoly_BLAKE2b`.
+This can be broken down into the following components:
```

Comment:
Yes, the plaintext version exchange was intended to act as an escape hatch in case we needed to move away from Noise for some reason, or if folx used I2P or Tor and didn't want the Noise stuff adding unnecessary cost to the transport.

I'm not sure if that's wise or not!